### PR TITLE
Add vsphere-csi 2.4.1 package

### PR DESCRIPTION
**Pull request by blc1996**
## What this PR does / why we need it

Will bump to GA image when CSI 2.4.1 GA is released

Add CSI 2.4.1 package (https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1450) that fixes the topology upgrade issue for users that are using beta multi-AZ feature with older releases.

Changes:
* Add a `useTopologyFeature` flag in values.yaml. If true, use `topology-category` label; if false, use `region` and `zone` label.
* User vSphere-CSI 2.4.1-rc.1 image
* Update README of the package

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Add vSphere-CSI 2.4.1 package
```

## Which issue(s) this PR fixes

Fixes: # vSphere-CSI multi-AZ topology upgrade issue

## Describe testing done for PR

template tests:
```
  region: null
  zone: null
  useTopologyFeature: false
```
->
```
  No labels
```
------
```
  region: region1
  zone: null
  useTopologyFeature: false
```
->
```
  [Labels]
   region = "region1"
```
------
```
  region: region1
  zone: zone1
  useTopologyFeature: false
```
->
```
  [Labels]
    region = "region1"
    zone = "zone1"
```
------
```
  region: null
  zone: null
  useTopologyFeature: true
```
->
```
  No labels
```
------
```
  region: null
  zone: zone1
  useTopologyFeature: true
```
->
```
  [Labels]
    topology-categories = "zone1"
```
------
```
  region: region1
  zone: zone1
  useTopologyFeature: true
```
->
```
  [Labels]
    topology-categories = "region1, zone1"
```

### CODE REVIEW

File: addons/packages/vsphere-csi/2.4.1/bundle/config/values.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  namespace: kube-system
+  clusterName: null
+  server: null
+  datacenter: null
+  publicNetwork: null
+  username: null
+  password: null
+  region: null
+  zone: null
+  useTopologyFeature: false
```
Comment (blc1996) [inline addons/packages/vsphere-csi/2.4.1/bundle/config/values.yaml:15]: `useTopologyFeature` added

File: addons/packages/vsphere-csi/2.4.1/bundle/config/vsphereconf.lib.txt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+password = "(@=values.vsphereCSI.password.replace("\\", "\\\\").replace("\"", "\\\"").replace("\t","\\t") @)"
+datacenters = "(@=values.vsphereCSI.datacenter @)"
+
+[Network]
+public-network = "(@=values.vsphereCSI.publicNetwork @)"
+(@ if values.vsphereCSI.region or values.vsphereCSI.zone: -@)
+
+[Labels]
+(@ # if useTopologyFeature has defined and true is passed in, csi config uses topology-categories. Otherwise still use region,zone -@)
+(@ if hasattr(values.vsphereCSI, "useTopologyFeature") and values.vsphereCSI.useTopologyFeature: -@)
```
Comment (blc1996) [inline addons/packages/vsphere-csi/2.4.1/bundle/config/vsphereconf.lib.txt:18]: Logic change in here
If `useTopologyFeature` is true, use `topology-categories`
else, use `region` and `zone`

File: addons/packages/vsphere-csi/2.4.1/bundle/.imgpkg/bundle.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: imgpkg.carvel.dev/v1alpha1
+kind: Bundle
+metadata:
+  name: vsphere-csi
+authors:
+  - name: Chethan Venkatesh
+    email: chethanv@vmware.com
+websites:
```
Comment (vijaykatam) [inline addons/packages/vsphere-csi/2.4.1/bundle/.imgpkg/bundle.yaml:8]: nitpick: update authors

File: addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            - name: "HTTPS_PROXY"
+              value: #@ values.vsphereCSI.https_proxy
+            - name: "NO_PROXY"
+              value: #@ values.vsphereCSI.no_proxy
+        #@ end
+        #@overlay/match by=overlay.subset({"name": "csi-provisioner"})
+        - args:
+            #@overlay/match by=lambda index, left, right: "--timeout=" in left, expects=1
+            - #@ "--timeout=" + values.vsphereCSI.provisionTimeout
+              #@ if values.vsphereCSI.region or values.vsphereCSI.zone:
```
Comment (vijaykatam) [inline addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml:53]: Should we use `useTopologyFeature` for adding these 2 arguments? Also indentation for if/end is not aligned.

Comment (blc1996) [reply addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml:53]: Does older case (region and zone are set, topologyFeature is false) require these flags?

Comment (danniel1205) [reply addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml:53]: IIUC, the enablement of `--feature-gates=Topology=true` is based on whether `values.vsphereCSI.region` or `values.vsphereCSI.zone` is passed in. `useTopologyFeature` has a confusing name, it actually tells whether to use `topology-categories` or `region, zone` in csi configs.

File: addons/packages/vsphere-csi/2.4.1/bundle/.imgpkg/bundle.yaml (hunk 3 above)
Comment (blc1996) [reply addons/packages/vsphere-csi/2.4.1/bundle/.imgpkg/bundle.yaml:10]: Updated.

File: addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml (hunk 4 above)
Comment (blc1996) [reply addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml]: Fixed the indentation.
Updated the flag name to `useTopologyCategories`

**Review by vijaykatam — APPROVED** [review]

**Review by danniel1205 — APPROVED** [review]

**Pull request merged**